components/shared/filters/modules/usePopularCollections.ts

```diff
@@ -0,0 +1,67 @@
+import { POPULAR_COLLECTIONS } from './constants'
+import { CollectionEntityMinimal } from '@/components/collection/utils/types'
+
+type QueryResult = {
+  value: {
+    collectionEntities: Collection[]
+  }
+}
+
+function handleResult(
+  collections: Collection[],
+  result: QueryResult[],
+  index: number
+): Collection[] {
+  const chain = Object.keys(POPULAR_COLLECTIONS)[index]
+  const newCollections =
+    result[index].value?.collectionEntities?.map((item) => ({
+      ...item,
+      owners: new Set(item.nfts.map((x) => x.currentOwner)).size,
+      chain,
+    })) || []
+  return collections
+    .concat(newCollections)
+    .sort((a, b) => a.meta.name.localeCompare(b.meta.name))
+}
+
+export const collectionArray = ref<Collection[]>([])
+
+export type Collection = CollectionEntityMinimal & {
+  owners: number
+  chain: string
+  meta: {
+    name: string
+  }
+  nfts: {
+    currentOwner: string
+  }[]
+}
+
+export const usePopularCollections = () => {
+  const collections = ref<Collection[]>([])
+  const resArr = ref<QueryResult[]>([])
+  const loadingMap = reactive<{ [key: string]: { value: boolean } }>({})
+
+  for (const [clientName, ids] of Object.entries(POPULAR_COLLECTIONS)) {
+    const { data, loading } = useGraphql({
+      queryName: 'collectionByIds',
+      clientName,
+      variables: { ids },
+    })
+    loadingMap[clientName] = loading
+    resArr.value.push(data)
+  }
+
+  // Aoid using Array as root value for reactive() as it cannot be tracked in watch() or watchEffect(). Use ref() instead. This is a Vue-2-only limitation.
+  watch(loadingMap, (val) => {
+    Object.keys(val).forEach((key, index) => {
+      if (!loadingMap[key].value) {
+        collections.value = handleResult(collections.value, resArr.value, index)
+        collectionArray.value = collections.value
+        delete loadingMap[key]
```

Comment:
1. don't use ```delete```
2. seems to me this will be quite easier if you create an object 
that looks like 
```js
{ [chain]: { 
    loading: boolean,
    data: queryResult
   }
 }
```
and just process the data of the corrosponding loading when it turns to ```false```

Reply:
@daiagi hmmm but in this way we may need an extra field to flag if this chain's data has been handled. wdyt?

Reply:
Why? 
You watch the loading, and when it's done you process the corrosponding data and push the results to the output array that the vue components use

Reply:
yes, i watch the `chainData` object, and there's a for each loop in the handler, if we do not add the flag it will process the data repeated🤔️  did i make sence?

```typescript
 watch(chainData, (val) => {
    Object.keys(val).forEach((chain, index) => {
      ...
  })
```

Reply:
Yeah I see your point.
I think it will result in cleaner simpler code
But it's just a suggestion...